dali/kernels/imgproc/warp_gpu.h

```diff
@@ -27,15 +27,23 @@ namespace dali {
 namespace kernels {
 
 /**
- * @remarks Assume HWC layout
+ * @brief Performs generic warping of a batch of tensors (on GPU)
+ *
+ * The warping uses a mapping functors to map destination coordinates to source
+ * coordinates and samples the source tensors at the resulting locations.
+ *
+ * @remarks
+ *  * Assumes HWC layout
+ *  * Output and input have same number of spatial dimenions
+ *  * Output and input have same number of channels and layout
  */
-template <typename _Mapping, int ndim, typename _OutputType, typename _InputType,
+template <typename _Mapping, int _spatial_ndim, typename _OutputType, typename _InputType,
```

Comment:
question: why not `spatial_dim` and get rid of the definition in line 45?

Reply:
Because I want it to be accessible as a dependent name `Kernel::spatial_dim`.